src/controller/ReadInteraction.h

```diff
@@ -64,13 +65,23 @@ CHIP_ERROR ReportAttribute(Messaging::ExchangeManager * exchangeMgr, EndpointId
     auto readClient =
         chip::Platform::MakeUnique<app::ReadClient>(engine, exchangeMgr, callback->GetBufferedCallback(), readParams.mReportType);
 
-    ReturnErrorOnFailure(readClient->SendRequest(readParams));
+    if (readClient->IsSubscriptionType())
+    {
+        err = readClient->SendAutoResubscribeRequest(std::move(readParams));
```

Comment:
It seems to me that the right contract should be, given that we std::move here, and touching the object after that is bad form:
1. If `SendAutoResubscribeRequest` is called at all, it guarantees that it will call `OnDeallocatePaths`.  That might be before it returns, on errors, or after OnDone if it succeeds.
2. SendAutoResubscribeRequest is the _only_ case that calls `OnDeallocatePaths`, since that's the only case when the consumer moved a ReadParams into the client.

Reply:
Updated the comments for SendAutoResubscribeRequest to reflect this point, OnDone is callled when the error happens or suceeds. OnDeallocatePaths is called inside OnDone